### PR TITLE
feat: Add storybook utils to labs common

### PR DESCRIPTION
**Pull request by vibdev**
## Summary

CKR has some really great internal tools for showing variations of components in Storybook. 

One of the great things about CKR is being able to build up or theme our own components based on base CKR components. 

However, it is arduous to test our variations. 

Well good news! Now you can get your very own storybook utils right here in CKR Labs Common.

## Checklist

- [x] branch has been rebased on the latest master commit
- [x] tests are changed or added
- [x] `yarn test` passes
- [x] all (dev)dependencies that the module needs is added to its `package.json`
- [x] code adheres to the [API & Pattern guidelines](../modules/docs/mdx/API_PATTERN_GUIDELINES.mdx)

### CODE REVIEW

File: modules/labs-react/common/lib/storybook-utils/ComponentStatesTable.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    paddingRight: 16,
+    textAlign: 'left',
+  },
+});
+
+export const ComponentStatesTable = ({
+  rowProps,
+  columnProps,
+  children,
+}: ComponentStatesTableProps) => {
```
Comment (vibdev) [inline modules/labs-react/common/lib/storybook-utils/ComponentStatesTable.tsx:45]: Changed this to a FC from a class. Also dropped the default props, I couldn't find any module actually using them, so figured it was better to just make them required.

File: modules/labs-react/common/lib/storybook-utils/permutateProps.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+export const permutateProps = (
```
Comment (vibdev) [inline modules/labs-react/common/lib/storybook-utils/permutateProps.ts:1]: split this out to its own file

File: utils/spec/tsconfig.json [hunk 3]
```diff
@@ -1,3 +0,0 @@
-{
-  "extends": "../../tsconfig.spec.json"
```
Comment (anicholls) [inline utils/spec/tsconfig.json:2]: Any reason you removed this guy?

Comment (vibdev) [reply utils/spec/tsconfig.json:2]: I removed the folder since there were no specs left

Comment (anicholls) [reply utils/spec/tsconfig.json:2]: Hah - thanks 😅

**Review by NicholasBoll — APPROVED** [review]
The only weird thing is seeing these components imported from a `common` location. A component state table could definitely be used in a production setting, but most likely would be used for testing. I wonder if `test-utils` or something would be a more appropriate location...

**Review by anicholls — APPROVED** [review]

**Comment by anicholls** [comment]
@NicholasBoll You mean adding a new sub-folder inside the labs module? I'm fine either way.

**Comment by vibdev** [comment]
I think common was chosen to align with the StaticStates export.

**Comment by NicholasBoll** [comment]
> @NicholasBoll You mean adding a new sub-folder inside the labs module? I'm fine either way.

Yes, but I'm fine if it is in common. I'm not sure we meant to export `StaticStates` in common, but I'm if it's there. It just looks weird if I'm importing and in the autocomplete is Storybook utility components and functions.

**Pull request merged**